gateway-js/src/__tests__/integration/configuration.test.ts

```diff
@@ -204,6 +206,62 @@ describe('gateway config / env behavior', () => {
     }
   });
 
+  describe('introspection headers', () => {
+    test('should use static headers', async () => {
+      let receivedHeaders;
+      mockSDLQueryFn(service, function reply() {
+        receivedHeaders = this.req.headers;
+        return [
+          200,
+          {
+            data: { _service: { sdl: print(service.typeDefs) } },
+          },
+        ];
```

Comment:
Not too happy with this, will think of something neater

Reply:
https://github.com/apollographql/federation/pull/607/commits/ec2d3fe40bce652fe4a1b28c4122e0ba832e415a